tests/snappi_tests/utils/metrics.py

```diff
@@ -0,0 +1,122 @@
+"""
+This file defines the classes receiving metrics from snappi tests and processing them.
+"""
+import logging
+import json
+import datetime
+import time
+
+from copy import deepcopy
+from pprint import pprint
+from typing import Dict, Final, List, Union
+
+# Only certain labels are allowed
+METRIC_LABEL_TESTBED: Final[str] = "testbed"
+METRIC_LABEL_TEST_BUILD: Final[str] = "os.version"
+METRIC_LABEL_TEST_CASE: Final[str] = "testcase"
+METRIC_LABEL_TEST_FILE: Final[str] = "test_file"
+METRIC_LABEL_TEST_JOBID: Final[str] = "job_id"
+METRIC_LABEL_DEVICE_ID: Final[str] = "device.id"            # device refers to the level of switch
+METRIC_LABEL_DEVICE_PORT_ID: Final[str] = "device.port.id"
+METRIC_LABEL_DEVICE_PSU_ID: Final[str] = "device.psu.id"
+METRIC_LABEL_DEVICE_QUEUE_ID: Final[str] = "device.queue.id"
+METRIC_LABEL_DEVICE_SENSOR_ID: Final[str] = "device.sensor.id"
+METRIC_LABEL_COMPONENT_MODEL: Final[str] = "model"          # component refers to the level below, i.e. parts used by a switch
+METRIC_LABEL_COMPONENT_SERIAL: Final[str] = "serial"
+METRIC_LABEL_CAST_DIRECTION: Final[str] = "cast_direction"  # unicast or multicast
+METRIC_LABEL_HARDWARE_REVISION: Final[str] = "hardware.revision"
+METRIC_LABEL_PRIORITY_GROUP: Final[str] = "priority_group"
+METRIC_LABEL_BUFFER_POOL: Final[str] = "buffer_pool"
```

Comment:
it will be better to have the name more... organized per feature. labels needs to be explicit, so they don't step on each other's toe in the dashboard, e.g.: 2 metrics might both having the name "model" - one for PSU, one for transceiver. So, this naming is not a good approach.

I would recommend to be:

```python
METRIC_LABEL_DEVICE_PSU_MODEL: Final[str] = "device.psu.model"          # component refers to the level below, i.e. parts used by a switch
METRIC_LABEL_DEVICE_PSU_SERIAL: Final[str] = "serial"
METRIC_LABEL_DEVICE_PG_ID: Final[str] = "device.pg.id"
METRIC_LABEL_DEVICE_BUFFER_POOL_ID: Final[str] = "device.buffer_pool.id"
```

I cannot really tell what are these below, but I bet you should now getting the idea and can make the adjustments by your own now.

```python
METRIC_LABEL_CAST_DIRECTION: Final[str] = "cast_direction"  # unicast or multicast
METRIC_LABEL_HARDWARE_REVISION: Final[str] = "hardware.revision"
```